### PR TITLE
Featured Items: Remove inline style defaults for color options

### LINKED ISSUE #7042: Featured Items: Inline style color defaults override Global Styles
## Summary

The text color defaults for the Featured Product and Featured Category blocks are set as inline styles. The problem is that the inline styles also override any changes to the block's Global Styles in the Site Editor.

## To reproduce

1. Ensure you have a block theme active (i.e., Twenty Twenty-Two).
2. Go to **Appearance > Editor (Beta)**.
3. Click on the **Global Styles** panel and select the **Blocks** section.
4. Scroll down to the Featured Product or Featured Category block and edit the text and the background Global Styles.
5. Notice that the background changes but the text remains white.

## Expected behavior

The text setting should also be adjustable when edited via Global Styles.

## Screenshots

| Global Styles Panel | Global Styles Block Setting |
| ------ | ----- |
|  CleanShot 2022-08-31 at 16 48 11  |  CleanShot 2022-09-01 at 14 31 05  |

## Additional context

The purpose for the inline default, as implemented in #6492, was to ensure the white text over the overlay, as some theme styles (i.e., Storefront) were potentially interfering.

Now that we have Block Themes and Global Styles becoming more prevalent, we should ensure that our blocks are editable via the controls offered. We can add an editor-specific style for added specificity but we do not want so much specificity that the Global Styles are overridden, as they are currently. IMO, the theme styles should be responsible to ensure they are not unintentionally overriding global settings.

cc @Aljullu and @sunyatasattva for your input, as you were both involved in #6489 and #6492.

### PR DESCRIPTION
**Pull request by danielwrobert**
This replaces the inline style in the Featured Product and Featured Category blocks with an editor-specific styling that has a bit more specificity - but not too much to where it overrides the Site Editor styles.

Fixes #7042 

### Screenshots

| Before | After |
| ------ | ----- |
|  CleanShot 2022-09-01 at 14 31 05  |  CleanShot 2022-09-01 at 14 35 45  |

### Testing

1. Ensure you have a block theme active (i.e., Twenty Twenty-Two).
2. Go to **Appearance > Editor (Beta)**.
3. Click on the **Global Styles** panel and select the **Blocks** section.
4. Scroll down to the **Featured Product** or **Featured Category** block and edit the text and the background Global Styles.
5. Confirm that both the text and background colors reflect the changes from the sidebar controls.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [x] Feature plugin
* [ ] Experimental

### Changelog

> Featured Item Blocks: remove inline default color so that custom colors from Global Styles are applied correctly.

### CODE REVIEW

**Review by Aljullu — COMMENTED** [review]
Thanks for your research in this tricky issue, @danielwrobert! I took an initial look at this PR, but it seems like the global style changes (made in the site editor) aren't reflected in the frontend. Can you reproduce as well?

Editor | Frontend
--- | ---
imatge | imatge

**Comment by danielwrobert** [comment]
@Aljullu no I can not reproduce what you're seeing. I'm getting the expected result:

| Editor | Frontend |
| ------ | ----- |
|  CleanShot 2022-09-01 at 14 35 45  |  Screen Shot 2022-09-02 at 8 42 42 AM  |

Do you happen to have the Gutenberg feature plugin active? That is a separate issue where the latest version seems to prevent Woo and Jetpack blocks from properly rendering Global Styles to the frontend.

File: assets/js/blocks/featured-items/featured-category/editor.scss
Comment (Aljullu) [inline assets/js/blocks/featured-items/featured-category/editor.scss]: If the purpose of this ruleset is only intended to fix #6489, I wonder if we should make it explicit in some way (maybe with a comment?).

Also, not blocking but wondering if instead of implementing this work-around here, we could update Storefront to avoid causing this issue. The problem seems to be that [this ruleset](https://github.com/woocommerce/storefront/blob/ed2530f09d447c8f94e39569a9bb3fbd38638d49/inc/customizer/class-storefront-customizer.php#L1070-L1072) has too much specificity and messes up the Featured Product/Category blocks, right? Maybe we could refactor that code from Storefront in some way? (I didn't really investigate that, just thinking out loud in case you have an idea, @danielwrobert :slightly_smiling_face: )

**Review by Aljullu — APPROVED** [review]
Oh, my bad, you are right. The issue was from Gutenberg (for reference, it's this one: https://github.com/WordPress/gutenberg/issues/40808). Disabling GB fixes it.

Confirming this PR fixes the issue.

File: assets/js/blocks/featured-items/featured-category/editor.scss
Comment (danielwrobert) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: > If the purpose of this ruleset is only intended to fix https://github.com/woocommerce/woocommerce-blocks/issues/6489, I wonder if we should make it explicit in some way (maybe with a comment?).

@Aljullu this still really doesn't fix #6489 on Storefront's side of things. I really only added this ruleset in to add a little bit more specificity to the styles in the editor, more generally, but it should still work the same way without this.

Thinking about it more, I'll go ahead and remove this ruleset from the editor stylesheet because it's not really consistent with everything else and this styling is intended for both the frontend and admin.

> Also, not blocking but wondering if instead of implementing this work-around here, we could update Storefront to avoid causing this issue.

Yes, that's 100% my thinking. IMO, the theme styles should be responsible to ensure they are not unintentionally overriding global (or other) settings. I noted this on the #7042 but should have reiterated my point here.

**So next steps:** I'll go ahead and remove the editor styles from this PR, as they're not really necessary, and merge. Then I'll make a note on #6489 (for posterity) and open a new issue on Storefront to address the problem you noted with the [overly specific ruleset](https://github.com/woocommerce/storefront/blob/ed2530f09d447c8f94e39569a9bb3fbd38638d49/inc/customizer/class-storefront-customizer.php#L1070-L1072) there.

Comment (Aljullu) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: I'm a bit confused, this PR seems to reintroduce #6489. It's not clear to me if that's a known issue or not.

imatge

> **So next steps:** I'll go ahead and remove the editor styles from this PR, as they're not really necessary, and merge. Then I'll make a note on #6489 (for posterity) and open a new issue on Storefront to address the problem you noted with the [overly specific ruleset](https://github.com/woocommerce/storefront/blob/ed2530f09d447c8f94e39569a9bb3fbd38638d49/inc/customizer/class-storefront-customizer.php#L1070-L1072) there.

It sounds good to me, but wondering if we should prioritize this fix in Storefront. Otherwise that would be a regression. :confused:

Comment (danielwrobert) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: > I'm a bit confused, this PR seems to reintroduce https://github.com/woocommerce/woocommerce-blocks/issues/6489. It's not clear to me if that's a known issue or not.

@Aljullu that's correct. The issue is that the way the initial fix for #6489 was handled (via inline style attributes) prevents the global styles from being applied in the editor - which can be more confusing for users, IMO.

The idea here is to revert back to the original color rules and then handle the issue in #6489 on the Storefront side. Generally, it seems better to handle theme-specific issues from the theme side, since we won't be able to account for what different themes do that may override our plugin styles.

I agree that this should be prioritized in Storefront. I am planning on getting that going today so it doesn't get put on the back-burner and the regression gets rolled out to our users.

I can even hold off on merging this until the Storefront PR is further along, if you think that'd be better?

Additionally, @sunyatasattva [suggested a broader conversation on P2](https://github.com/woocommerce/woocommerce-blocks/issues/7042#issuecomment-1237917157) around how we prioritize styles overall, which I plan to post as a follow-up as well.

Comment (Aljullu) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: Your explanation makes sense, sorry if I misunderstood it before. Your next steps look good to me.

> I can even hold off on merging this until the Storefront PR is further along, if you think that'd be better?

I will leave this up to you. Storefront has a shorter release process, so I think we are good if we merge this PR as-is and work on a fix in SF afterwards.

Comment (danielwrobert) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: Sounds good, thanks for the follow-up! I've created [an Issue on Storefront](https://github.com/woocommerce/storefront/issues/2036) and assigned myself to address ASAP.

Comment (sunyatasattva) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: Yea just to chime in here as well, I think it is unclear what is the expected hierarchy between styles, and who should be concerned of being compatibile with what. Should themes make sure they are compatible with plugins or viceversa? I think we should probably come up with a scalable solution for these cases in general. 🤔

Comment (danielwrobert) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: @sunyatasattva I agree! I'll kick off a broader P2 discussion around this today.

For this particular PR, the default color styles preventing the Global Styles from working seems a bit more confusing for a user, as opposed to allowing a theme's defaults to override and impact the color contrast of the Featured Items blocks.

I'm totally open to being convinced otherwise and I'm also not opposed to closing this PR out if it doesn't seem like a good way to go!

Overall, it's a bit of a tricky situation where we want the Global Styles to function as expected while also preventing themes from overriding the default colors with poor contrast.

I'm actually wondering if it's acceptable to move forward with removing this default and not worry about the overriding color contrast issue in Storefront? In that scenario the user still has control to adjust the text color and the background overlay, as needed.

Alternatively, we could also remove the dark background overlay default as well and leave _all_ of the color adjustments to the user when they are using the block. I'm not certain this is a good option, just throwing it out there. I'd love to hear your thoughts!

Comment (sunyatasattva) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: I'm 100% with you: definitely default color preventing Global Styles is a mistake. Not providing a sensible default… I'd be against that. How all these variables interact with each other… that's where it gets really tricky and it's definitely a philosophical question at this point :)

Comment (danielwrobert) [reply assets/js/blocks/featured-items/featured-category/editor.scss]: > I'm 100% with you: definitely default color preventing Global Styles is a mistake.

Since this seems to be the general consensus, I will go ahead and merge this PR to resolve that problem.

We can then continue to explore the best route forward for the text overriding issue with Storefront in [this discussion](https://github.com/woocommerce/storefront/pull/2039).

**Pull request merged**
